### ISSUE TITLE
PIU.HyperOCR.darkToLightTimer fail

### ISSUE THREAD

**Issue by oceancolorcoder** [nasa/HyperCP#214]
Mac, M99MA.

```
Process Single Level: /Users/daurin/Projects/HyperPACE/field_data/HyperSAS/KORUS/L1BQC/1614402_L1BQC.hdf - SUCCESSFUL
Process Single Level
ProcessL2: /Users/daurin/Projects/HyperPACE/field_data/HyperSAS/KORUS/L1BQC/1614402_L1BQC.hdf
stationsEnsemblesReflectance
**************Cannot do cubic spline interpolation, length of datasets < 3
Binning datasets to ensemble time interval.
Traceback (most recent call last):
  File "/Users/daurin/GitRepos/HyperCP/Main.py", line 546, in multi2Clicked
    self.processMulti(2)
  File "/Users/daurin/GitRepos/HyperCP/Main.py", line 540, in processMulti
    Controller.processFilesMultiLevel(self.outputDirectory,fileNames, calibrationMap, flag_Trios)
  File "/Users/daurin/GitRepos/HyperCP/Source/Controller.py", line 815, in processFilesMultiLevel
    Controller.processSingleLevel(pathOut, fp, calibrationMap, 'L2', flag_Trios)
  File "/Users/daurin/GitRepos/HyperCP/Source/Controller.py", line 736, in processSingleLevel
    root = Controller.processL2(root,outFilePath)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daurin/GitRepos/HyperCP/Source/Controller.py", line 449, in processL2
    node = ProcessL2.processL2(root,station)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daurin/GitRepos/HyperCP/Source/ProcessL2.py", line 2302, in processL2
    if not ProcessL2.stationsEnsemblesReflectance(node, root,station):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daurin/GitRepos/HyperCP/Source/ProcessL2.py", line 2193, in stationsEnsemblesReflectance
    if not ProcessL2.ensemblesReflectance(node, sasGroup, referenceGroup, ancGroup, uncGroup, esRawGroup,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daurin/GitRepos/HyperCP/Source/ProcessL2.py", line 1311, in ensemblesReflectance
    stats = instrument.generateSensorStats("SeaBird",
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daurin/GitRepos/HyperCP/Source/ProcessInstrumentUncertainties.py", line 77, in generateSensorStats
    output[sensortype] = self.lightDarkStats([rawData[sensortype]['LIGHT'],
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daurin/GitRepos/HyperCP/Source/ProcessInstrumentUncertainties.py", line 1765, in lightDarkStats
    lightData[k][x] -= darkData[k][x]
                       ~~~~~~~~~~~^^^
IndexError: list index out of range
```

**Comment by oceancolorcoder** [comment]
Failure to interpolate L1AQC datasets carried forward to L2 for PIU purposes derives from very small files. A filter has been set in ProcessL2 to reject any ensemble shorter than 60 s, which does prevent the crashes like above. However, it does not avoid the interpolation error, because this is done before ensemble slicing. 

Still unclear why darks are interpolated to light timestamps in the "raw" dataset at L2 for PIU.

**Comment by oceancolorcoder** [comment]
Darks are interpolate to lights at L2 in order to reproduce the dark correction prior to uncertainty updates for ensemble variability.

This issue should be resolved with 0454f9135bd4f3ba316eb2768fefc68cfc05c96a.
